### ISSUE TITLE
Types within an interface not properly checked when interface is IMyInterface<any>

### ISSUE THREAD

**Issue by Roaders** [microsoft/TypeScript#36627]
**TypeScript Version:**  3.7.5

**Search Terms:**

**Code**

```ts
class MyClass{
    public on = "string";
    public two = 2
    public three = true;
}

interface IReplacements<T>{
    original: T,
    replacements: Partial<T>
}

function doReplacements(callback: () => IReplacements<any>[]) {
    
}

doReplacements(() => {
    return [
        {
            original: new MyClass,
            replacements: {
                notDefined: "this should error" // should throw error as notDefined not on MyClass
            }
        }];
})
```

**Expected behavior:**
`not defined` should fail to compile as it is not defined on `MyClass`

**Actual behavior:**
Compiels fine as interface is of type any

**Playground Link:** 
http://www.typescriptlang.org/play/?ssl=25&ssc=3&pln=2&pc=1#code/FAYwNghgzlAECyBPAwpGBvYtuwA4FcAjMASxFgHsA7WAXlgCIoAXAJxKoHMGBuLHAsTKxmAdwp1YAJn7ZBpcswAWrAKarJbfKr4BfYMA7NVrAGYQQGgJIAlVbkiWAtqqrMoAHgAqAPkw5Kdk4OCDAALlgvABpZWDUHC1UXNygIgAUIVmYSUO8fYH1gU3wqEGzqWAATCjsE51d3AAoQULBCCwBrCMaASjofWFt7RySGzwgqRB8AbQBdPv8cAoNq2pHkpt7+2EXsNWZ8Vhpp2IDdgIuKIJDw2CpVUQQUNCgYi-e44cSN1J3Tj4CVAozAAIqpTBxVJUIgxlCQ4FAlBR8GBKrATKwrgxYAB6HGwRHI1EiFQUR4Yq6waB3YFgiH3NFA5iUGhIVDQKD-d76AG6WZ6HoGEDUFifOqjNwRIbin4eNkvAb0c5XEjBKihCL3R7yjm9N44eLrMYRc6A2ngyHQxhwhFIlFoimsbF42DmEhgODU1QAD1wqjKUNiuhihSAA

The issue is that I want an an array of `IReplacements` but the only way of sepcifying an array is by saying `IReplacements<any>[]`. I want this to be a replacement of any type but I still want the `replacements` to be properly type checked as `Partial` of `T`

**Comment by Roaders** [comment]
OK, I managed to fix this but it's slightly obscure IMHO

```
interface IReplacements<T>{
    original: T,
    replacements: Partial<T>
}

type toReplacements<T extends any[]> = {
    [I in keyof T]: IReplacements<T[I]>
}

function doReplacements<T extends any[]>(callback: () => toReplacements<T>) {
    
}

doReplacements(() => {
    return [
        {
            original: new MyClass,
            replacements: {
                notDefined: "this should error" // should throw error as notDefined not on MyClass
            }
        }];
})
```

http://www.typescriptlang.org/play/#code/FAYwNghgzlAECyBPAwpGBvYtuwA4FcAjMASxFgHsA7AU1gF5YAiKAFwCcSqBzJgbiw4CxMrFYB3Cg1gAmQdmGlyrABbsadRh3w0BAX2DAurGuwBmEEHQCSAJRq5IVgLY0qrKAB4AKgD5MOJSc3FwQYABcsN4ANPKw6o6WNK7uUJEAChDsrCRhPr7ABsCsiLh0rBT2iS5uHj6wNAAeJlQAJnAQVIgA2gC6vtIBON3WsFywANY0iBRmUb2Rdg5OybVe3iP9hYZm+FQgOdSwrZXLSSl13g3Nbu2wnT39ABQgYWCElhORTwCUDAMVKorC7rXx-IbYbbAE5A85rJ6-f6wCHxGisfDsKiwbpxQIowKBCjBUIRWC0cQIFBoKCxAl01HVVapSL4+mBKgUVgAERoZi4NFakSYqhIcCgKgo+DArQa7HYRKYsAA9ErYOLJdKxGoKBTTPL2Pc4Bzubz+TLjZQsUhUNAoLj6QY2XpevofoYQNQ2AzgbVFrCaqlPNbqQNGCiiSQQlQwpFyZSbTBfrScAkfczkfbsMaeXzaILmCKxRKpTK9QrlaqLCQwB04E0ygcBXE9LEikA
